R/capture-requests.R

```diff
@@ -260,24 +260,46 @@ dbGetInfoConTrace <- quote({
   }
 })
 
-dbGetInfoResultTrace <- quote({
-  thing <- returnValue()
-  # TODO: would this be better if we traced the methods individually?
-  if (inherits(dbObj, c("MariaDBResult", "PqResult", "SQLiteResult"))) {
-    hash <- hash(dbObj@sql)
-  } else if (inherits(dbObj, "OdbcResult")) {
-    hash <- hash(dbObj@statement)
+#' Extract a hash from a (result) object
+#'
+#' This function should generally not be used, but must be exported for the
+#' query recording function to work properly
+#'
+#' @param obj the DBI result object to hash
+#'
+#' @return hash for the object
+#' @export
+#' @keywords internal
+extract_hash_from_object <- function(obj) {
```

Comment:
This is a bit nitpicky, but what about calling this function `hash_object` or `hash_db_object` that's shorter and slightly more direct (we usually do extract some info and hash it, but the we don't extract the hash itself)